supertokens_python/recipe/session/recipe_implementation.py

```diff
@@ -234,126 +276,267 @@ async def validate_claims_in_jwt_payload(
-            return None
-        anti_csrf_token = get_anti_csrf_header(request)
-        if anti_csrf_check is None:
-            anti_csrf_check = normalise_http_method(request.method()) != "get"
 
-        log_debug_message(
-            "getSession: Value of doAntiCsrfCheck is: %s", str(anti_csrf_check)
-        )
-        new_session = await session_functions.get_session(
-            self,
-            access_token,
-            anti_csrf_token,
-            anti_csrf_check,
-            get_rid_header(request) is not None,
-        )
-        if "accessToken" in new_session:
-            access_token = new_session["accessToken"]["token"]
+        try:
+            anti_csrf_token = get_anti_csrf_header(request)
+            do_anti_csrf_check = anti_csrf_check
 
-        if access_token is None:
-            raise Exception("Should never come here")
-        session = Session(
-            self,
-            access_token,
-            new_session["session"]["handle"],
-            new_session["session"]["userId"],
-            new_session["session"]["userDataInJWT"],
-        )
+            if do_anti_csrf_check is None:
+                do_anti_csrf_check = normalise_http_method(request.method()) != "get"
+            if request_transfer_method == "header":
+                do_anti_csrf_check = False
+            log_debug_message(
+                "getSession: Value of doAntiCsrfCheck is: %s", do_anti_csrf_check
+            )
 
-        if "accessToken" in new_session:
-            session.new_access_token_info = new_session["accessToken"]
+            new_session = await session_functions.get_session(
+                self,
+                access_token,
+                anti_csrf_token,
+                do_anti_csrf_check,
+                get_rid_header(request) is not None,
+            )
 
-        log_debug_message("getSession: Success!")
-        request.set_session(session)
-        return request.get_session()
+            access_token_string = access_token.raw_token_string
+            if new_session.get("accessToken") is not None:
+                # We set the expiration to 100 years, because we can't really access the expiration of the refresh token everywhere we are setting it.
+                # This should be safe to do, since this is only the validity of the cookie (set here or on the frontend) but we check the expiration of the JWT anyway.
+                # Even if the token is expired the presence of the token indicates that the user could have a valid refresh
+                # Setting them to infinity would require special case handling on the frontend and just adding 10 years seems enough
+                set_token(
+                    self.config,
+                    response,
+                    "access",
+                    new_session["acessToken"]["token"],
+                    int(datetime.now().timestamp()) + 3153600000000,
+                    request_transfer_method,
+                )
```

Comment:
you should not have access to the response variable here - so can't use this method...